File: Project Euler/problem12.py

def generateTriangle(i):
	return i*(i+1)/2
def countDivisors(num):
	count=0
	i=1
	while True:
		if num%i==0:
			count=count+1
		if i>num/2:
			break
		i=i+1
	return count+1
i=100
while True:
	n=generateTriangle(i)
	count=countDivisors(n)
	print count
	if count>500:
		print n
		break
	i=i+1

File: Project Euler/problem12.c
#include <stdio.h>

int generateTriangle(int i){
  return i*(i+1)/2;
}
int divisors(int n){
  int limit=n;
  int count=0;
  int i;
  for(i=1;i<limit;++i){
    if (n%i==0){
      limit=(int)n/i;
      count++;
    }
  }
  return count*2;
}
int main(int argc,char **argv){
 long i=100;
  while (1){
    long n=generateTriangle(i);
    int count=divisors(n);
    if (count>500){
      printf("%ld\n",n);
      break;
    }
    i++;
  }
  return 0;
}
